### PR TITLE
test: attempt to quote python interpreter path

### PR DESCRIPTION
**Pull request by compnerd**
Attempt to properly quote the python interpreter path in the lit configuration.  This
was quoted in some places but not others.  This attempts to homogenise the path quoting.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
CC: @tachoknight @drexin @yln

File: test/lit.cfg
Comment (yln) [inline test/lit.cfg:381]: This is the actual change as discussed in #32571.

**Review by yln — APPROVED** [review]
LGTM.  Please wait for one more approval from a Swift maintainer.

**Review by drexin — COMMENTED** [review]
LGTM, thanks!

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

**Pull request merged**
